src/ripple/app/ledger/impl/ConsensusImp.cpp

```diff
@@ -177,15 +177,22 @@ ConsensusImp::takePosition (int seq, std::shared_ptr<SHAMap> const& position)
     }
 }
 
-void
-ConsensusImp::visitStoredProposals (
-    std::function<void(LedgerProposal::ref)> const& f)
+std::vector <std::shared_ptr <LedgerProposal>>
+ConsensusImp::getStoredProposals (uint256 const& prevLedger)
 {
-    std::lock_guard <std::mutex> _(lock_);
 
-    for (auto const& it : storedProposals_)
-        for (auto const& prop : it.second)
-            f(prop);
+    std::vector <std::shared_ptr <LedgerProposal>> ret;
+
+    {
+        std::lock_guard <std::mutex> _(lock_);
+
+        for (auto const& it : storedProposals_)
+            for (auto const& prop : it.second)
+                if (prop->isPrevLedger (prevLedger))
```

Comment:
Consider removing `isPrevLedger` and changing this to: `prop->getPrevLedger() == prevLedger`. You can cherry-pick 41bf613105f724cd26f44f769560dab0616b3995 if you want.